src/recordpage.cpp

```diff
@@ -834,6 +863,29 @@ void RecordPage::setError(const QString& message)
     ui->applicationRecordErrorMessage->setVisible(!message.isEmpty());
 }
 
+void RecordPage::addKonsoleWidget()
+{
+    if (m_service) {
+        const auto tail = QStandardPaths::findExecutable(QLatin1String("tail"));
+
+        if (!tail.isEmpty()) {
+            if (m_konsolePart) {
+                ui->recordOutputBox->layout()->removeWidget(m_konsolePart->widget());
+                m_konsolePart->deleteLater();
+            }
+            m_konsolePart = m_service->createInstance<KParts::ReadOnlyPart>(this);
+            m_konsoleFile = new QTemporaryFile(this);
```

Comment:
check the validity of the kpart early instead of blindly using it and its casted variant:
```
if (!m_konsolePart) {
    return;
}
const auto terminalInterface = qobject_cast<TerminalInterface*>(m_konsolePart);
if (!terminalInterface) {
    qWarning("konsole kpart doesn't implement terminal interface");
    delete m_konsolePart;
    m_konsolePart = nullptr;
    return;
}
```